service-tests/bundlephobia.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const Joi = require('joi')
+const ServiceTester = require('./runner/service-tester')
+
+const t = new ServiceTester({
+  id: 'bundlephobia', title: 'NPM package bundle size',
+})
+
+const formats = {
+  A: '/bundlephobia/:type/:package.:format',
+  B: '/bundlephobia/:type/:package/:version.:format',
+  C: '/bundlephobia/:type/@:scope/:package.:format',
+  D: '/bundlephobia/:type/@:scope/:package/:version.:format',
+}
+
+const withoutErrorsMin = {
+  A: '/min/preact.json',
+  B: '/min/preact/8.2.7.json',
+  C: '/min/@cycle/core.json',
+  D: '/min/@cycle/core/7.0.0.json',
+}
+
+const withoutErrorsGzip = {
+  A: '/gzip/preact.json',
+  B: '/gzip/preact/8.2.7.json',
+  C: '/gzip/@cycle/core.json',
+  D: '/gzip/@cycle/core/7.0.0.json',
+}
+
+const withoutErrorSizesMin = {
+  A: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  B: '8.94 kB',
+  C: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  D: '3.51 kB',
+}
+
+const withoutErrorSizesGzip = {
+  A: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  B: '3.58 kB',
+  C: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  D: '1.23 kB',
+}
+
+const noExistPackages = {
+  A: '/min/some-no-exist.json',
+  B: '/min/some-no-exist/1.0.0.json',
+  C: '/min/@some-no-exist/some-no-exist.json',
+}
+
+Object.keys(formats).forEach(format => {
+  const withoutErrorMin = withoutErrorsMin[format]
+  const withoutErrorSizeMin = withoutErrorSizesMin[format]
+
+  const withoutErrorGzip = withoutErrorsGzip[format]
+  const withoutErrorSizeGzip = withoutErrorSizesGzip[format]
+
+  const noExistPackage = noExistPackages[format]
+
+  if (typeof withoutErrorMin === 'string') {
+    t.create(`Format '${formats[format]}' to get the minified bundle size`)
+      .get(withoutErrorMin)
+      .expectJSONTypes(Joi.object().keys({ name: 'minified size', value: withoutErrorSizeMin }))
+  }
+
+  if (typeof withoutErrorGzip === 'string') {
+    t.create(`Format '${formats[format]}' to get the gzipped bundle size`)
+      .get(withoutErrorGzip)
+      .expectJSONTypes(Joi.object().keys({ name: 'gzip size', value: withoutErrorSizeGzip }))
+  }
+
+  if (typeof noExistPackage === 'string') {
+    t.create(`Format '${formats[format]}' where it doesn't exist`)
+      .get(noExistPackage)
+      .expectJSON({
+        name: 'minified size',
+        value: 'PackageNotFoundError',
```

Comment:
**minified size | package not found** or even **bundlephobia | package not found** would be more stylistically consistent with our error badges.

Reply:
These error codes are returned by the bundlephobia service, and there are multiple such types - `EntryPointError` | `PackageNotFoundError` | `VersionMismatchError` etc.  I'd prefer to keep them the same to enable easier debugging when issues are filed.

Reply:
I think it's sufficient to re-run the failing request when an issue is filed. One of the goals of Shields is to provide consistent and concise badges, which applies to error messages too.